src/formatDate.js

```diff
@@ -7,8 +7,51 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const SEPARATOR_FROM_FORMAT = fromFormat[3];
+  const SEPARATOR_TO_FORMAT = toFormat[3];
+
+  const SHORT_YEAR_FORMAT = 'YY';
+  const LONG_YEAR_FORMAT = 'YYYY';
+
+  const MAX_YEAR_NEW_CENTURY = 30;
+  const NEW_CENTURY = 20;
+
+  const dateArr = date.split(SEPARATOR_FROM_FORMAT);
+  const formatedDateArr = [];
+
+  const dateLength = dateArr.length;
+
+  for (let i = 0; i < dateLength; i++) {
+    for (let j = 0; j < dateLength; j++) {
+      if (toFormat[i] === fromFormat[j]) {
+        formatedDateArr.push(dateArr[j]);
+        break;
+      }
+
+      if (
+        toFormat[i] === SHORT_YEAR_FORMAT &&
+        fromFormat[j] === LONG_YEAR_FORMAT
+      ) {
+        formatedDateArr.push(dateArr[j].slice(2));
+      }
+
+      if (
+        toFormat[i] === LONG_YEAR_FORMAT &&
+        fromFormat[j] === SHORT_YEAR_FORMAT
+      ) {
+        if (dateArr[j] < MAX_YEAR_NEW_CENTURY) {
```

Comment:
When comparing string numbers, the comparison should be done as integers. Otherwise, '10' would be considered less than '2' because string comparison is based on lexicographical order. You should convert `dateArr[j]` to an integer before the comparison.